substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/ThreadingSupportImpl.java

```diff
@@ -241,6 +241,7 @@ private void invokeCallback() {
 
     @Override
     public void registerRecurringCallback(long interval, TimeUnit unit, RecurringCallback callback) {
+        // TODO: We should make this method uninterruptible. How to allocate Timer for each thread?
```

Comment:
I don't think that it is necessary to make this uninterruptible. It should be fine to call `registerRecurringCallback` at a later point in time (e.g., `beforeThreadRun`) where it is no longer necessary to execute uninterruptible code.

However, it might be necessary to refactor the logic in `RecurringCallbackTimer` as another recurring callback could already be registered.

Reply:
Removed.